### ISSUE TITLE
Character movement will cause the ground item text to shake

### ISSUE THREAD

**Issue by HoneSundan** [bolrog/d2dx#105]
Is this a bug?

**Comment by bolrog** [comment]
Which game version is this?

**Comment by jaydee2k** [comment]
Have the same issue with 1.13d. Only with Motion Prediction turned on.
There is no shaking without MoP.

**Comment by bolrog** [comment]
Thanks, I'll have a look at this.

**Comment by HoneSundan** [comment]
1.13c

**Comment by bolrog** [comment]
I don't see any shaking when testing this.

You are using the latest D2DX I assume?

**Comment by jaydee2k** [comment]
Yes I use the latest D2DX. It's only shaking wit MoP on. As I said in the other bug report I play with 155 hz but it doesn't matter.
It shakes from 60-155hz on every setting.

**Comment by tomasz1986** [comment]
I'm not sure if this is the same issue, but I have managed to record the shaking text. Notice how "BOLTS" keeps changing its position while the character is moving.

https://youtu.be/VZLwmmeP8xo

**Comment by bolrog** [comment]
Strange... I wish I saw this happen on my machine. Hmm.

**Comment by tomasz1986** [comment]
> Strange... I wish I saw this happen on my machine. Hmm.

I have just done some testing, trying to reproduce the issue, and it seems that this happens when there are other items on the ground, but located right at the edge of the screen. Basically, those other items only show and disappear when the character is being moved. While this is happening, the original item's label keeps changing its position.

You can observe this in the video above, where some other items keep popping up at the top of the screen. When that happens, the bolts' text also moves.

**Comment by bolrog** [comment]
That's helpful! I'll see what I can find.

There are some cases where the position will "jump", which is when the game reorders the hover-text boxes. Basically, I will smoothly move the boxes when the movement is small (say, < 8 pixels), but if the game moves a text box by more than that, I assume it wants to rearrange things e.g. because a new item landed on the ground. Hopefully it's just a matter of tweaking this movement threshold.

**Comment by jaydee2k** [comment]
The text is shaking during gameplay constantly while moving for me. It has nothing to do with other items laying on the ground in my case. It's more an effect like the text is sliding on ice. I will se if I can shoot a video from it.

**Comment by jaydee2k** [comment]
Heres a video from my end:

[https://youtu.be/K0p4yqoeo50](url)

**Comment by bolrog** [comment]
Oh, now I get it. Yes, that's a known issue. The text doesn't move exactly at the same rate the screen does, so it lags behind a bit.

Hopefully I can address this soon.

**Comment by KierTaylor** [comment]
Just to add to this, it happens in v1.09d too (very extreme on my end in comparison to the video from jaydee)

**Comment by bolrog** [comment]
If it's extreme ("shaking" as opposed to just gliding a bit) then it's a bug. I thought I'd fixed it in 1.09 though. Hm.

**Comment by nebulus-aversus** [comment]
I have this as well and I'm kinda getting motion sick from it - weird since I never get that.

1.13c, motion prediction on.
